google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/TableInfo.java

```diff
@@ -130,6 +131,8 @@ public Table apply(TableInfo tableInfo) {
     public abstract TableInfo build();
 
     public abstract Builder setEncryptionConfiguration(EncryptionConfiguration configuration);
+
+    public abstract Builder setRequirePartitionFilter(Boolean requirePartitionFilter);
```

Comment:
This is an API breaking change that requires a major version bump per semver.  If this were non-abstract with a default implementation that would not be so.

Reply:
@elharo please take a look on this existing PR Ref https://github.com/googleapis/google-cloud-java/pull/6321/files#diff-2814f0d886d621f57ac6eaaabd329765R127-R134, same thing introduced before and it did not break any API change.

Reply:
This breaks all existing subclasses. It looks like an incompatible API change. So does the one you cite. What am I missing?

Reply:
@shollyman do you think if users are extending TableInfo.Builder ? i think it's most unlikely that's why i introduced setter.

Reply:
@elharo TableInfo.Builder is extended only once within the client.

Reply:
The class is non-final, public, and has an implicit public constructor. In hindsight, I suspect that was a mistake but we're stuck with it now until the next major version bump. We've made a commitment to our users not to break the API until then so we need to find another way around this. There are several. The cleanest and simplest is probably a default implementation that makes the method non-abstract.

Reply:
If I'm parsing elharo's comment correctly, the issue is we're starting with a new abstract implementation that all implementors must deal with, and that may break existing subclassing.

The default implementation (in the backend) treats this as false when not specified.  However, an alternate default implementation choice here may be to surface a NotImplementedException to signal to third parties about missing functionality.  Which feels like the better solution here to you @elharo?

Reply:
Can the default implementation simply set the field in the usual way?

Reply:
Discussed w Seth - from BQ product perspective we've been adhering to using abstract class without default, hence we will make it different here. Although point brought up here is completely valid.

Reply:
will add an empty default implementation.